src/lib/operators/join_hash.hpp

```diff
@@ -52,6 +52,19 @@ class JoinHash : public AbstractJoinOperator {
     size_t radix_bits{0};
     // Initially, the left input is the build side and the right side is the probe side.
     bool left_input_is_build_side{true};
+
+    // Due to the used Bloom filters, the number of actually joined tuples can significantly differ from the sizes of
+    // the input tables. To enable analyses of the Bloom filter efficiency, we store the number of values that were
+    // eventually materialized; i.e., "input_row_count - filtered_values_by_Bloom_filter".
+    size_t build_side_materialized_value_count{0};
+    size_t probe_side_materialized_value_count{0};
+
+    // In build(), the Bloom filter potentially reduces the distinct values in the hash table (i.e., the size of the
+    // hash table) and the number of rows (in case of non-semi/anti* joins).
+    // Note, depending on the order of materialization, build_side_materialized_value_count is not necessarily equal to
+    // build_side_position_count (see order of materialization in hash_join.cpp).
+    size_t hash_tables_distinct_value_count{0};
+    size_t hash_tables_position_count{0};
```

Comment:
Why don't you pass the optional all the way to the consumer? The whole idea was that the consumer can identify the difference between 0 and N/A.

Reply:
(No need to re-review once this is addressed)

Reply:
Hmm, any ideas how to do that cleaner than this?
```
auto hash_tables_position_count = size{0};
    for (const auto& hash_table : hash_tables) {
      if (!hash_table) continue;

      _performance_data.hash_tables_distinct_value_count += hash_table->distinct_value_count();
      _performance_data.hash_tables_position_count += hash_table->position_count() ? *hash_table->position_count() : 0;
    }
    if (hash_tables_position_count > 0) {
      _performance_data.hash_tables_position_count = hash_tables_position_count;
    }
```

Reply:
````_performance_data.hash_tables_position_count =
  _performance_data.hash_tables_position_count.value_or(0) + *position_count;
```
?

Also, just noticed that it should say "IF hash table stores", not "when".